src/main/java/uk/nhs/tis/sync/job/PostFundingSyncJob.java

```diff
@@ -0,0 +1,97 @@
+package uk.nhs.tis.sync.job;
+
+import com.transformuk.hee.tis.tcs.api.enumeration.Status;
+import com.transformuk.hee.tis.tcs.service.model.Post;
+import com.transformuk.hee.tis.tcs.service.model.PostFunding;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Set;
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import lombok.extern.slf4j.Slf4j;
+import net.javacrumbs.shedlock.core.SchedulerLock;
+import org.apache.commons.collections4.CollectionUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.jmx.export.annotation.ManagedOperation;
+import org.springframework.jmx.export.annotation.ManagedResource;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+import uk.nhs.tis.sync.model.EntityData;
+
+/**
+ * Job for updating funding status for posts. This class extends the
+ * PersonDateChangeCaptureSyncJobTemplate and provides functionality to synchronize post funding
+ * status based on specified criteria.
+ */
+@Component
+@ManagedResource(objectName = "sync.mbean:name=PostFundingSyncJob",
+    description = "Job for updating funding status for posts")
+@Slf4j
+public class PostFundingSyncJob extends PostFundingSyncJobTemplate<Post> {
```

Comment:
I prefer to naming this job `PostFundingStatusSyncJob` as it's specific for fundingStatus.